Fix warning on network errors in perl bindings (Bug #422699).
It's actually a common case where an error structure doesn't have an
auxilliary message string, so we shouldn't expect one.
From upstream r1146834, included in 1.7.x.


--- a/subversion/bindings/swig/perl/native/Core.pm
+++ b/subversion/bindings/swig/perl/native/Core.pm
@@ -581,7 +581,8 @@ sub expanded_message {
 
 		my $error_message = $svn_error->strerror();
 		while ($svn_error) {
-		    $error_message .= ': ' . $svn_error->message();
+		    my $msg = $svn_error->message();
+		    $error_message .= ": $msg" if $msg;
 				$svn_error = $svn_error->child();
 		}
 		return $error_message;
